database/migrations/2021_08_06_184400_refactor_album_model.php

```diff
@@ -0,0 +1,823 @@
+				'altitude' => $photo->altitude,
+				'imgDirection' => $photo->img_direction,
+				'location' => empty($photo->location) ? null : $photo->location,
+				'taken_at' => $photo->taken_at,
+				'taken_at_orig_tz' => $photo->taken_at_orig_tz,
+				'type' => $photo->type,
+				'filesize' => $photo->filesize,
+				'checksum' => $photo->checksum,
+				'live_photo_short_path' => $photo->live_photo_short_path,
+				'live_photo_content_id' => $photo->live_photo_content_id,
+				'live_photo_checksum' => $photo->live_photo_checksum,
+			]);
+		}
+
+		$sizeVariants = DB::table('size_variants_tmp')->lazyById();
+		foreach ($sizeVariants as $sizeVariant) {
+			DB::table('size_variants')->insert([
+				'id' => $sizeVariant->id,
+				'photo_id' => $sizeVariant->photo_id,
+				'size_variant' => $sizeVariant->size_variant,
+				'short_path' => $sizeVariant->short_path,
+				'width' => $sizeVariant->width,
+				'height' => $sizeVariant->height,
+			]);
+		}
+	}
+
+	/**
+	 * Upgrades the configuration of default ordering to the new column names.
+	 */
+	private function upgradeConfig(): void
+	{
+		DB::table('configs')
+			->where('key', '=', 'sorting_Photos_col')
+			->update(['type_range' => 'id|taken_at|title|description|is_public|is_starred|type']);
+		DB::table('configs')
+			->where('key', '=', 'sorting_Photos_col')
+			->where('value', '=', 'public')
+			->update(['value' => 'is_public']);
+		DB::table('configs')
+			->where('key', '=', 'sorting_Photos_col')
+			->where('value', '=', 'star')
+			->update(['value' => 'is_starred']);
+		DB::table('configs')
+			->where('key', '=', 'sorting_Albums_col')
+			->update(['type_range' => 'id|title|description|is_public|max_taken_at|min_taken_at|created_at']);
+		DB::table('configs')
+			->where('key', '=', 'sorting_Albums_col')
+			->where('value', '=', 'public')
+			->update(['value' => 'is_public']);
+	}
+
+	/**
+	 * Downgrades the configuration of default ordering to the new column names.
+	 */
+	private function downgradeConfig(): void
+	{
+		DB::table('configs')
+			->where('key', '=', 'sorting_Photos_col')
+			->update(['type_range' => 'id|taken_at|title|description|public|star|type']);
+		DB::table('configs')
+			->where('key', '=', 'sorting_Photos_col')
+			->where('value', '=', 'is_public')
+			->update(['value' => 'public']);
+		DB::table('configs')
+			->where('key', '=', 'sorting_Photos_col')
+			->where('value', '=', 'is_starred')
+			->update(['value' => 'star']);
+		DB::table('configs')
+			->where('key', '=', 'sorting_Albums_col')
+			->update(['type_range' => 'id|title|description|public|max_taken_at|min_taken_at|created_at']);
+		DB::table('configs')
+			->where('key', '=', 'sorting_Albums_col')
+			->where('value', '=', 'is_public')
+			->update(['value' => 'public']);
+	}
+
+	/**
+	 * A helper function that allows to drop an index if exists.
+	 *
+	 * @param Blueprint $table
+	 * @param string    $indexName
+	 */
+	private function dropIndexIfExist(Blueprint $table, string $indexName)
```

Comment:
Shouldn't this method (and the two below as well) be called `[...]IfExists`? :smiley:

Reply:
You are right. I fixed it.